tests/components/syncthru/test_config_flow.py

```diff
@@ -0,0 +1,136 @@
+    aioclient_mock.get(
+        re.compile("."),
+        text="""
+{
+\tstatus: {
+\tstatus1: "  Sleeping...   "
+\t},
+\tidentity: {
+\tserial_num: "000000000000000",
+\t}
+}
+        """,
+    )
+
+
+async def test_show_setup_form(hass):
+    """Test that the setup form is served."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}, data=None
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+
+async def test_already_configured_by_url(hass, aioclient_mock):
+    """Test we match and update already configured devices by URL."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    udn = "uuid:XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
+    MockConfigEntry(
+        domain=DOMAIN,
+        data={**FIXTURE_USER_INPUT, CONF_NAME: "Already configured"},
+        title="Already configured",
+        unique_id=udn,
+    ).add_to_hass(hass)
+    mock_connection(aioclient_mock)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}, data=FIXTURE_USER_INPUT,
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result["data"][CONF_URL] == FIXTURE_USER_INPUT[CONF_URL]
+    assert result["data"][CONF_NAME] == FIXTURE_USER_INPUT[CONF_NAME]
+    assert result["result"].unique_id == udn
+
+
+async def test_syncthru_not_supported(hass):
+    """Test we show user form on unsupported device."""
+    with patch.object(SyncThru, "update", side_effect=ValueError):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+            data=FIXTURE_USER_INPUT,
+        )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {CONF_URL: "syncthru_not_supported"}
+
+
+async def test_unknown_state(hass):
+    """Test we show user form on unsupported device."""
+    with patch.object(SyncThru, "update", return_value=mock_coro()), patch.object(
+        SyncThru, "is_unknown_state", return_value=True
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+            data=FIXTURE_USER_INPUT,
+        )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {CONF_URL: "unknown_state"}
+
+
+async def test_success(hass, aioclient_mock):
+    """Test successful flow provides entry creation data."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    mock_connection(aioclient_mock)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}, data=FIXTURE_USER_INPUT,
+    )
```

Comment:
You need to patch `async_setup_entry` in your integration and `await hass.async_block_till_done()`

Reply:
Done, although I'm not quite sure why is that. Test seems to work fine without it, and although according to output, mocking `async_setup_entry` appears to avoid writing to the entity registry, a config entry gets written anyway so it's not all without writes in the first place.

Reply:
It's a race condition. Sometimes it will end up setting things up during your test teardown, causing a flaky test.

Reply:
Ok, thanks for the info :+1: